aperturedb/Connector.py

```diff
@@ -366,7 +366,7 @@ def _query(self, query, blob_array = [], try_resume=True):
         data = query_msg.SerializeToString()
 
         # this is for session refresh attempts
-        tries = 0
+        tries = -1
```

Comment:
this means there will be self.config.retry_max_attempts + 1 retries. 

I think what you want is to keep `tries = 0` and move line 408 to 414.

Reply:
my dilemma is on the variable name. The first is technically not a retry, but this is the only place for retry mechanism. 

Let me think more . 

I though about moving 408 to 414 but that'd be 3 connection attempts, 2 of which are actual retries.

Reply:
we could call it connection attempts instead of retry and be done with it. Thoughts?